Hw5/Task2/Catalog.cs

```diff
@@ -0,0 +1,53 @@
+using System.Collections.ObjectModel;
+using System.Text.RegularExpressions;
+
+public class Catalog
+{
```

Comment:
Better to implement separate class for Isbn, move all checks and conversions to it, and use     Dictionary<Isbn, Book> dictionary = new();